classes/02class/exercises/c02-network07/frdvo/README.md

```diff
@@ -0,0 +1,226 @@
+    "AssociationState": {
+        "State": "associated"
+    }
+}
+
+#Associate public-b
+$ aws ec2 associate-route-table --route-table-id  rtb-03acbacd49d5fcb60 --subnet-id subnet-0dbb2205880974071
+{
+    "AssociationId": "rtbassoc-0bcb8050a42183c95",
+    "AssociationState": {
+        "State": "associated"
+    }
+}
+
+#Associate private-b
+$ aws ec2 associate-route-table --route-table-id rtb-024a69525b35bfded --subnet-id subnet-0b9349fa56d76afb7
+{
+    "AssociationId": "rtbassoc-03b9d5bf5974f1ac8",
+    "AssociationState": {
+        "State": "associated"
+    }
+}
+
+#Associate public-c
+$ aws ec2 associate-route-table --route-table-id rtb-03acbacd49d5fcb60 --subnet-id subnet-01f6ba32065d77e19
+{
+    "AssociationId": "rtbassoc-081b79bf9e4cce40e",
+    "AssociationState": {
+        "State": "associated"
+    }
+}
+
+#Associate private-c
+$ aws ec2 associate-route-table --route-table-id rtb-024a69525b35bfded --subnet-id  subnet-074a113f80e846338
+{
+    "AssociationId": "rtbassoc-047cf390184c684ab",
+    "AssociationState": {
+        "State": "associated"
+    }
+}
+
+```
+
+- Commands for creating the following routes:
+
+|route table|destination|target|
+|-|-|-|
+|devopsacademy-rt-public|0.0.0.0|`devopsacademy-igw`|
+|devopsacademy-rt-private|0.0.0.0|`devopsacademy-ngw`|
+
+```
+#devopsacademy-rt-public
+$ aws ec2 create-route --route-table-id rtb-03acbacd49d5fcb60 --destination-cidr-block 0.0.0.0/0 --gateway-id igw-0116304685cc6f00b
+{
+    "Return": true
+}
+
+#devopsacademy-rt-private
+aws ec2 create-route --route-table-id rtb-024a69525b35bfded --destination-cidr-block 0.0.0.0/0 --nat-gateway-id nat-0ad38ee136b255aab
+{
+    "Return": true
+}
+
+```
+
+- Answer the following questions:
+  - How did you accomplish both instructions?
+    ```
+    Create and public and private route table and assign to the c02-network04 subnets. I also recreate the deleted NAT gateway
+    ```
+
+  - Why did you add 0.0.0.0 route to the IGW on public route table?
+    ```
+    0.0.0.0 means all internet, if the IP is outside the network, IGW you try to reach the IP on Internet using the instance public IP
+    ```
+
+  - Why did you add 0.0.0.0 route to the NGW on private route table?
+    ````
+    0.0.0.0 means all internet, if the IP is outside the network, NGW you try to reach the IP on Internet using it's own IP and NAT technologies.
+    ````  
+    
+  - What is the difference of IGW and NGW?
+    ```
+    IGW is to use when you have a instance with public IP that should receive inbound connections. NAT is for instances without external IP that should connect to internet, without expose to the internet.
+    ```
+    
+  - Can you delete the destination route to your VPC network? Why?
+    ```
+    No, this route table is to communicate with other machines inside the VPC.
+    ```
+    
+  - What happens if you do not associate your route table with any subnets?
+    ```
+    It will be associate with the main route table
```

Comment:
Not exatcly. We don't need to associate subnets with route tables, and route tables don't need to have subnets associated to it.

A route table without subnets won't route any requests from subnets (there is no use for it, basically).

Can you fix this one?

Reply:
Sure, thanks for the explanation.